### PR TITLE
baguette

### PR DESCRIPTION
**Pull request by Daxz0**

### CODE REVIEW

File: plugins/Denizen/scripts/baguette.dsc [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+baguette_eat:
+    type: world
+    debug: false
+    events:
+        on player consumes baguette:
+            - define item <player.item_in_hand>
+            - if <[item].flag[bites]> <= 0:
+                - stop
+            - inventory flag slot:hand bites:-:1
```
Comment (acikek) [inline plugins/Denizen/scripts/baguette.dsc]: Use the `--` data action

File: plugins/Denizen/scripts/baguette.dsc [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+            - narrate <[item].flag[bites]>
```
Comment (acikek) [inline plugins/Denizen/scripts/baguette.dsc]: Debug narrate left in?

File: plugins/Denizen/scripts/baguette.dsc [hunk 3]
```diff
@@ -0,0 +1,27 @@
+baguette:
+    type: item
+    display name: <&f>Baguette
+    material: bread
+    flags:
+        bites: 4
+    lore:
+    - <&7>Its so long that it
```
Comment (acikek) [inline plugins/Denizen/scripts/baguette.dsc]: `Its` -> `It's` (repeated on line 27)

**Review by acikek — CHANGES_REQUESTED** [review]

File: plugins/Denizen/scripts/baguette.dsc [hunk 4]
```diff
@@ -16,12 +17,24 @@ baguette:
 
 baguette_eat:
     type: world
-    debug: false
+    debug: true
```
Comment (acikek) [inline plugins/Denizen/scripts/baguette.dsc]: Stray debug

File: plugins/Denizen/scripts/baguette.dsc [hunk 5]
```diff
@@ -16,12 +17,24 @@ baguette:
...
+            - determine passively cancelled
+            - choose <[item].flag[bites]>:
+                - case 4:
+                    - define amount 2
+                - case 3:
+                    - define amount 4
+                - case 2:
+                    - define amount 6
+                - case 1:
+                    - define amount 8
```
Comment (acikek) [inline plugins/Denizen/scripts/baguette.dsc]: Use a map lookup

**Review by acikek — CHANGES_REQUESTED** [review]

File: plugins/Denizen/scripts/baguette.dsc [hunk 6]
```diff
@@ -0,0 +1,37 @@
+baguette:
+    type: item
+    display name: <&f>Baguette
+    material: bread
+    flags:
```
Comment (funkychicken493) [inline plugins/Denizen/scripts/baguette.dsc:5]: I don't think this type of item should be stackable due to it's 'reusable' nature, if you had a stack of these and used one then the following script would appear to break. Consider adding a uniqueifier to the flags to keep these from stacking.

**Review by funkychicken493 — CHANGES_REQUESTED** [review]

File: plugins/Denizen/scripts/baguette.dsc [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+        bites: 4
+        allergy: true
```
Comment (funkychicken493) [inline plugins/Denizen/scripts/baguette.dsc]: I'm confused, what does the allergy flag do?

File: plugins/Denizen/scripts/baguette.dsc [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+            - determine passively cancelled
+            - definemap map:
+                4: 2
+                3: 4
+                2: 6
+                1: 8
+            - define amount <[map].get[<[item].flag[bites]>]>
+            - feed amount:<[amount]> saturation:<[item].flag[bites].add[4]>
+            - inventory flag slot:hand bites:--
+            - inventory adjust slot:hand "lore:<&7>Its so long that it<n><&7>takes 5 bites to eat.<n><&7><n><&7>Bites left: <&8><[item].flag[bites]>"
```
Comment (funkychicken493) [inline plugins/Denizen/scripts/baguette.dsc]: Small nitpick, but this repeats lore from the item information, so if you were to change your lore on the item then it wouldn't change here. I recommend pulling the lore from the item script like <script[baguette].data_key[lore]> and then removing the last line, so you may replace it with an updated amount.

File: plugins/Denizen/scripts/baguette.dsc [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+    type: world
+    debug: false
+    events:
+        on player consumes baguette:
+            - define item <player.item_in_hand>
+            - if <[item].flag[bites]> <= 0:
+                - feed amount:17 saturation:<[item].flag[bites].add[4]>
+                - stop
+            - determine passively cancelled
+            - definemap map:
```
Comment (funkychicken493) [inline plugins/Denizen/scripts/baguette.dsc]: Bad definition* name.

File: plugins/Denizen/scripts/baguette.dsc [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+            - determine passively cancelled
+            - definemap bite_count:
+                4: 2
+                3: 4
+                2: 6
+                1: 8
+            - define amount <[bite_count].get[<[item].flag[bites]>]>
+            - feed amount:<[amount]> saturation:<[item].flag[bites].add[4]>
+            - inventory flag slot:hand bites:--
+            - inventory adjust slot:hand "lore:<&7>Its so long that it<n><&7>takes 5 bites to eat.<n><&7><n><&7>Bites left: <&8><[item].flag[bites]>"
```
Comment (funkychicken493) [inline plugins/Denizen/scripts/baguette.dsc]:
```suggestion
            - inventory adjust slot:hand "lore:<[item].lore.get[1].to[-2].separated_by[<n>]><n><&7>Bites left: <&8><[item].flag[bites]>"
```

File: plugins/Denizen/scripts/baguette.dsc [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+            - define item <player.item_in_hand>
+            - if <[item].flag[bites]> <= 0:
+                - feed amount:17 saturation:<[item].flag[bites].add[4]>
+                - stop
+            - determine passively cancelled
+            - definemap bite_count:
+                4: 2
+                3: 4
+                2: 6
+                1: 8
```
Comment (creaturesofhabit) [inline plugins/Denizen/scripts/baguette.dsc]: this can be 10-(2x)

File: plugins/Denizen/scripts/baguette.dsc [hunk 12]
```diff
@@ -0,0 +1,39 @@
...
+            - define amount <[bite_count].get[<[item].flag[bites]>]>
+            - feed amount:<[amount]> saturation:<[item].flag[bites].add[4]>
+            - inventory flag slot:hand bites:--
+            - inventory adjust slot:hand "lore:<[item].lore.get[1].to[-2].separated_by[<n>]><n><&7>Bites left: <&8><[item].flag[bites]>"
```
Comment (creaturesofhabit) [inline plugins/Denizen/scripts/baguette.dsc]: this lore change seems really hacky? It looks like you're just trying to change line 4? If so, just use <[item].lore.set[(your stuff)].at[4]>

**Review by creaturesofhabit — CHANGES_REQUESTED** [review]

File: plugins/Denizen/scripts/baguette.dsc (hunk 12 above)
Comment (Daxz0) [reply plugins/Denizen/scripts/baguette.dsc]: funky did it

Comment (creaturesofhabit) [reply plugins/Denizen/scripts/baguette.dsc]: funky's solution is equally bad, but this wasn't it. In no case do you need to be rewriting the lore in it's entirety. Both of you seem to be forgetting that lore is a list and can be edited per-item

**Comment by Daxz0** [comment]
oops forgot to put the naming

File: plugins/Denizen/scripts/baguette.dsc [hunk 13]
```diff
@@ -0,0 +1,31 @@
...
+    type: world
+    debug: false
+    events:
+        on player consumes baguette:
+            - define item <player.item_in_hand>
+            - if <[item].flag[bites]> <= 0:
+                - feed amount:17 saturation:<[item].flag[bites].add[4]>
+                - stop
+            - determine passively cancelled
+            - define amount <element[10].sub[<element[2].mul[<[item].flag[bites]>]>]>
```
Comment (creaturesofhabit) [inline plugins/Denizen/scripts/baguette.dsc]:
```suggestion
            - define amount <element[10].sub[<[item].flag[bites].mul[2]>]>
```
efficiency

**Review by creaturesofhabit — APPROVED** [review]

**Review by funkychicken493 — APPROVED** [review]

**Review by acikek — APPROVED** [review]

**Pull request merged**